sw/airborne/modules/system_identification/pprz_chirp.h

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) Joost Meulenbeld
+ *
+ * This file is part of paparazzi
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+/**
+ * @file "modules/system_identification/pprz_chirp.c"
+ * @author Joost Meulenbeld
+ *
+ * Mathematical implementation of the chirp
+ * A "chirp" or frequency sweep is a sine wave with in time increasing frequency, and can be
+ * used for system identification purposes. This registers a broad frequency spectrum.
+ *
+ * Chirps can be made with frequency increasing linearly or exponentially with time. This is set using an argument in
+ * the chirp_init method. The latter one is better for system identification, according to [2].
+ *
+ * The time length of the chirp is best put at a minimum of 4 / f_minimum such that low-frequency effects
+ * can be correctly discovered.
+ *
+ * The fade_in argument determines if the chirp fades in with increasing amplitude at
+ * constant frequency (the lowest frequency). This makes it much easier to maintain the UAV around the trim position.
+ *
+ * Usage example:
+ * // Initialize a chirp between 1 and 5 Hz during 2 seconds (for sys. id. use minimum of 4 * 1/f0)
+ * struct chirp_t chirp;
+ * chirp_init(&chirp, 1, 5, 2, get_current_time());
+ *
+ * // During flight loop, add chirp value to stick input
+ * while (chirp_is_running(chirp, get_current_time())) {
+ *     control[YAW] = stick_control + chirp_update(chirp, get_current_time());
+ * }
+ *
+ * [1] https://en.wikipedia.org/wiki/Chirp for a derivation of the linear chirp
+ * [2] Aircraft and Rotorcraft System Identification, 2nd edition by M. Tischler for a derivation of exponential chirp
+ */
+
+#ifndef PPRZ_CHIRP_H
+#define PPRZ_CHIRP_H
+
+#include <math.h>
+#include <std.h>
+#include <stdbool.h>
+#include <stdlib.h>
```

Comment:
only `"std.h` should be needed here, move the rest to C file (if needed)

Reply:
done